app/validators/routing/types.py

```diff
@@ -51,7 +51,9 @@ def resolve_value_source_json_type(value_source, answers_with_context):
         return ANSWER_TYPE_TO_JSON_TYPE[answer_type]
 
     if source == "list":
-        return LIST_SELECTOR_TO_JSON_TYPE[value_source["selector"]]
+        if selector := value_source.get("selector"):
+            return LIST_SELECTOR_TO_JSON_TYPE[selector]
+        return TYPE_ARRAY
```

Comment:
We should add a test for this scenario. You can update the existing test named `test_validate_value_sources`. It tests with a selector, just need to add without a selector.

Reply:
This is still outstanding, the test you added is valid, but this should also test with the full source in `test_validate_value_sources` but without a selector.